### PR TITLE
Re-write CollectionService unit tests with AutoFixtures

### PR DESCRIPTION
**Pull request by hantatsang**
Partial work for #419

### CODE REVIEW

File: test/Core.Test/AutoFixture/CollectionFixtures.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using AutoFixture;
+using Bit.Core.Test.AutoFixture.Attributes;
+using Bit.Core.Test.AutoFixture.OrganizationFixtures;
+
+namespace Bit.Core.Test.AutoFixture
+{
+    internal class CollectionDefaultIdAutoDataAttribute : CustomAutoDataAttribute
+    {
+        public CollectionDefaultIdAutoDataAttribute() : base(new SutProviderCustomization(), new Organization { CollectionId = default(Guid)})
```
Comment (cscharf) [inline test/Core.Test/AutoFixture/CollectionFixtures.cs:10]: teeny formatting nit ('cause my eye will stick there the whole time 😉 )
```suggestion
        public CollectionDefaultIdAutoDataAttribute() : base(new SutProviderCustomization(), new Organization { CollectionId = default(Guid) })
```

File: test/Core.Test/Services/CollectionServiceTests.cs [hunk 2]
```diff
@@ -1,184 +1,166 @@
...
+            var utcNow = DateTime.UtcNow;
+
+            await sutProvider.Sut.SaveAsync(collection);
+
+            await sutProvider.GetDependency<ICollectionRepository>().Received().ReplaceAsync(collection);
+            await sutProvider.GetDependency<IEventService>().Received()
+                .LogCollectionEventAsync(collection, EventType.Collection_Updated);
+            Assert.True(collection.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
+            Assert.Equal(collection.CreationDate, creationDate);
+            Assert.True(collection.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
```
Comment (cscharf) [inline test/Core.Test/Services/CollectionServiceTests.cs:66]: Was this assertion intended to be different here or duplicated? Perhaps asserting the revision date is no longer = to the creation date?

**Review by cscharf — COMMENTED** [review]
I _think_ this looks good for the most part aside from the 2 small things I saw, but would love for @MGibson1 to take a look and provide any feedback or final approval.

File: test/Core.Test/Services/CollectionServiceTests.cs (hunk 2 above)
Comment (hantatsang) [reply test/Core.Test/Services/CollectionServiceTests.cs:66]: I accidentally duplicated it. There's already an assertion for `collection.CreationDate` to not have been changed so I think removing this line should be sufficient.

**Review by MGibson1 — CHANGES_REQUESTED** [review]
Sorry for the delay on review -- I've been moving and super busy.

I only have the one real concern with `MaxCollections` everything else is pretty nit-picky. Nice work, I love that you expanded these tests while bringing them up to the new framework 🥳

File: test/Core.Test/AutoFixture/CollectionFixtures.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Bit.Core.Test.AutoFixture
+{
+    internal class CollectionDefaultIdAutoDataAttribute : CustomAutoDataAttribute
+    {
+        public CollectionDefaultIdAutoDataAttribute() : base(new SutProviderCustomization(), new Organization { CollectionId = default(Guid) })
+        { }
+    }
+
+    internal class CollectionNonDefaultIdAutoDataAttribute : CustomAutoDataAttribute
```
[the remark this answers is not in the document]
Comment (hantatsang) [reply test/Core.Test/AutoFixture/CollectionFixtures.cs]: Yep, totally agree 😄

File: test/Core.Test/AutoFixture/OrganizationFixtures.cs [hunk 4]
```diff
@@ -15,6 +15,7 @@ namespace Bit.Core.Test.AutoFixture.OrganizationFixtures
     public class Organization : ICustomization
     {
         public bool UseGroups { get; set; }
+        public Guid CollectionId { get; set; }
```
[the remark this answers is not in the document]
Comment (hantatsang) [reply test/Core.Test/AutoFixture/OrganizationFixtures.cs]: Please let me know if you're happy with the change

**Review by MGibson1 — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**